protoc-gen-go/generator/generator.go

```diff
@@ -2252,6 +2258,12 @@ func (g *Generator) generateMessage(message *Descriptor) {
 			})
 		}
 
+		if field.GetOptions().GetDeprecated() {
+			// Generate deprecation comments for getters as well.
+			g.P("// Deprecated: ", mname, " should not be used by new code.")
+			g.P("// Underlying field is marked as [deprecated=true] in ", g.file.Name)
+			g.P("//")
```

Comment:
Is the 2 lines above needed? The end users may not care about the proto definition here.

Reply:
Is it needed? I don't know. I thought it would be useful to explain why a field is deprecated, but the only reason that it is deprecated in because of the proto definition. I don't have a strong opinion either way.

Reply:
Done.